sim/common/mop/trinkets_phase_4_54.go

```diff
@@ -701,4 +693,145 @@ func init() {
 		buffAuraID:       148897,
 		buffedStat:       stats.Intellect,
 	})
+
+	// Time-Lost Artifact
+	// Your melee and ranged attacks have a chance to grant 3647 haste for 20 sec.
+	// (Proc chance: 20%, 50s cooldown)
+	core.NewItemEffect(103678, func(agent core.Agent, state proto.ItemLevelState) {
+		character := agent.GetCharacter()
+
+		aura := character.NewTemporaryStatsAura(
+			"Winds of Time",
+			core.ActionID{SpellID: 148447},
+			stats.Stats{stats.HasteRating: core.GetItemEffectScaling(103678, 1.56799995899, state)},
+			time.Second*20,
+		)
+
+		triggerAura := character.MakeProcTriggerAura(core.ProcTrigger{
+			Name:               "Time-Lost Artifact Trigger",
+			Callback:           core.CallbackOnSpellHitDealt,
+			Outcome:            core.OutcomeLanded,
+			ProcMask:           core.ProcMaskMeleeOrMeleeProc | core.ProcMaskRangedOrRangedProc,
+			ICD:                time.Second * 50,
+			ProcChance:         0.2,
+			RequireDamageDealt: true,
```

Comment:
In general Melee trinkets do not have this flag: As in all ToT rppm Str/Agi trinkets can proc off Pummel for example which is flagged as a "Melee Special"